contracts/bridge/Bridge.sol

```diff
@@ -0,0 +1,98 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+interface IERC20 {
+    function balanceOf(address owner) external view returns (uint256);
+
+    function allowance(address owner, address spender)
+        external
+        view
+        returns (uint256);
+
+    function approve(address spender, uint256 value) external returns (bool);
+
+    function transfer(address to, uint256 value) external returns (bool);
+
+    function transferFrom(
+        address from,
+        address to,
+        uint256 value
+    ) external returns (bool);
+}
+
+contract Bridge {
+    address public admin;
+    address public relayer;
+    IERC20 public bholdusToken = IERC20(0xf4275FD572398215ea31e630832C9652E59051E9);
+
+    struct TransferInfo {
+        address from;
+        bytes32 to;
+        uint256 amount;
+        uint16 target_chain;
+        uint256 service_fee;
+    }
+
+    uint32 next_outbound_transfer_id = 0;
+    uint32 next_confirmed_outbound_transfer_id = 0;
+    uint32 next_inbound_transfer_id = 0;
+    uint256 service_fee = 10;
+
+    event TransferInitiated(uint256 indexed transfer_id, address from, bytes32 to, uint256 amount, uint256 target_chain);
+    event TokensReleased(uint256 indexed transfer_id, bytes32 from, address to, uint256 indexed amount);
+
+    mapping(uint256 => TransferInfo) _transferInfo;
+
+    constructor(
+        address _admin,
+        address _relayer
+    ){
+        admin = _admin;
+        relayer = _relayer;
+    }
+
+    function initiateTransfer(bytes32 to, uint256 amount, uint16 target_chain) public payable{
+        require(msg.value > 0, "Require fee to execute transaction");
```

Comment:
The condition is `msg.value == service_fee`

Reply:
https://github.com/Bholdus/bholdus-chain/blob/main/pallets/bridge/native-transfer/src/lib.rs#L205